modules/dnn/src/tensorflow/tf_importer.cpp

```diff
@@ -1468,6 +1468,8 @@ void TFImporter::populateNet(Net dstNet)
             int end_mask = getLayerAttr(layer, "end_mask").i();
             for (int i = 0; i < num; ++i)
             {
+                if (ends.at<int>(i) <= 0)
+                    ends.at<int>(i) -= 1;
```

Comment:
According to https://www.tensorflow.org/api_docs/python/tf/strided_slice:

> If the ith bit of begin_mask is set, begin[i] is ignored and the fullest possible range in that dimension is used instead. end_mask works analogously, except with the end range.

Can you please show the values of `ends` for new test?

Reply:
```
attr {
    key: "end_mask"
    value {
      i: 9
    }
  }

attr {
    key: "value"
    value {
      tensor {
        dtype: DT_INT32
        tensor_shape {
          dim {
            size: 4
          }
        }
        tensor_content: "\000\000\000\000\377\377\377\377\377\377\377\377\000\000\000\000"
      }
    }
  }
```
so end mask = 1 0 0 1 (i.e. 9)
and end is [0, -1, -1, 0]

Reply:
Ok, got it. That means ends are excluded from range: `[begin, end)`. Thanks!

So it's better to change condition to `if (ends.at<int>(i) < 0)`. Zero value should not be processed.